src/main/java/book/store/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package book.store.onlinebookstore.service.impl;
+
+import book.store.onlinebookstore.dto.order.CreateOrderRequestDto;
+import book.store.onlinebookstore.dto.order.OrderDto;
+import book.store.onlinebookstore.dto.order.UpdateOrderRequestDto;
+import book.store.onlinebookstore.dto.order.item.OrderItemDto;
+import book.store.onlinebookstore.exception.EntityNotFoundException;
+import book.store.onlinebookstore.mapper.OrderItemMapper;
+import book.store.onlinebookstore.mapper.OrderMapper;
+import book.store.onlinebookstore.model.CartItem;
+import book.store.onlinebookstore.model.Order;
+import book.store.onlinebookstore.model.OrderItem;
+import book.store.onlinebookstore.model.ShoppingCart;
+import book.store.onlinebookstore.repository.order.OrderRepository;
+import book.store.onlinebookstore.repository.order.item.OrderItemRepository;
+import book.store.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import book.store.onlinebookstore.repository.user.UserRepository;
+import book.store.onlinebookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Your shopping cart is empty"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        Order newOrder = initNewOrder(requestDto, userId);
+        orderRepository.save(newOrder);

```

Comment:
if you have a proper cascade - you'll be able to save order at the end, so that all order items are saved as well

Reply:
It doesn't work propperly, I'll write directly to ask some questions